vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2505,11 +2661,8 @@ protected void removeColumnLayer(ColumnLayer layer) {
         }
         layer.getColumns().forEach(column -> {
             Element parent = column.getElement().getParent();
-            int insertIndex = parent.indexOfChild(column.getElement());
-            parent.insertChild(insertIndex,
-                    ((ColumnGroup) column).getChildColumns().stream()
-                            .map(HasElement::getElement)
-                            .toArray(Element[]::new));
+            parent.appendChild(((ColumnGroup) column).getChildColumns().stream()
+                    .map(HasElement::getElement).toArray(Element[]::new));
```

Comment:
This change fixes the issue mentioned in https://github.com/vaadin/flow-components/pull/5990#discussion_r1472865414 but it causes another type of regression when you remove a row with joined cells:

https://github.com/vaadin/flow-components/assets/1222264/25e45708-87c9-4e14-82d1-b25ec2532d0a

```java
@Route(value = "vaadin-grid/beangridpage")
public class BeanGridPage extends Div {

    private int id = 0;

    public BeanGridPage() {
        Grid<Person> grid = new Grid<>(Person.class);
        grid.setItems(new Person("Jorma", 2018), new Person("Jarvi", 33));
        grid.setItemDetailsRenderer(
                LitRenderer.<Person> of("<div>${item.name} ${item.age}</div>")
                        .withProperty("name", Person::getFirstName)
                        .withProperty("age", Person::getAge));
        add(grid);
        grid.getColumns().forEach((col) -> col.setId("c"+id++));
        
        var row1 = grid.prependHeaderRow();
        grid.getColumns().forEach((col) -> row1.getCell(col).setText("r1 " + col.getId().get()));

        var row2 = grid.prependHeaderRow();
        grid.getColumns().forEach((col) -> row2.getCell(col).setText("r2 " + col.getId().get()));
        
        row2.join(grid.getColumns().get(1), grid.getColumns().get(2)).setText("r2 join");

        row2.join(grid.getColumns().get(5), grid.getColumns().get(6), grid.getColumns().get(7)).setText("r2 join 2");

        var cols = new ArrayList<>(grid.getColumns());
        Collections.reverse(cols);
        grid.setColumnOrder(cols);


        add(new Button("Remove row", event -> {
            grid.removeHeaderRow(row2);
        }));
    }

}
```

Reply:
I reverted the change and will submit a separate issue for the column ordering problems.